firebase-firestore/src/main/java/com/google/firebase/firestore/model/value/ObjectValue.java

```diff
@@ -57,6 +60,31 @@ public int typeOrder() {
     return TYPE_ORDER_OBJECT;
   }
 
+  /** Recursively extracts the FieldPaths that are set in this ObjectValue. */
+  public FieldMask getFieldMask() {
```

Comment:
Why did the way mutations calculate field masks need to change?

Reply:
Before, we used the mutation's FieldMask to compute the baseMutation, which caused us to over-select fields. With this PR, each mutation returns the precise set of field it cares about in an ObjectValue, but we still need to provide the field mask to the PatchMutation.